### ISSUE TITLE
Not able to load simple jpg's on WIN 10

### ISSUE THREAD

**Issue by bdrichards** [libvips/nip2#76]
As shown in image below, NIP2 will not load any images on WIN 10.
Using most current install.
Why are these jpgs not readable? They perfectly in every other app, they are small 256x256 pixel "tiles" that I want to "join" them (no blending necessary)

![nip2_2019-03-07_9-49-43](https://user-images.githubusercontent.com/18133188/53977743-90dedf80-40be-11e9-8f22-4ed4fd19ad0b.jpg)

**Comment by jcupitt** [comment]
Hello @bdrichards,

That's odd, it should work. I'll test tonight, if I can get my kids off the PC.

**Comment by bdrichards** [comment]
Thanks,
My WIN 10 SurfaceBook runs just fine, but it doesn't have the umphhh to handle the join operation.
The system that refuses to recognize the JPG's is a dell Precision 7720 ("mobile workstation") with 64Gb RAM,   a RadeonPro WX 7100 graphics card, and 1Tb SSD.
Could there be graphics driver issues? What about administrative rights, or lack thereof, interfering? I have set the properties for NIP2 to "Run as Administrator"

**Comment by jcupitt** [comment]
I tried and it's working fine for me. Could you try loading one of the examples? File / Open examples, then `1_point_mosaic`, then `1_point_mosaic.ws`.

I don't think a graphics driver would break jpg loading, so I'd rule that out.

It doesn't need admin rights -- you can install and run as an ordinary user.

Some other piece of software on your PC might be clashing with it. I would try making a new user and running it there. Also check and maybe reset your `PATH`.

**Comment by bdrichards** [comment]
The example images also could not be read with the example workspace. Also, I think I remember that VIPS itself couldn't find images from a command prompt test.  Sooo, does that hint at a "path" issue?
See the image below, the thumbnails are blank:
![np2-1pt_mosaic](https://user-images.githubusercontent.com/18133188/54049881-aae9f200-4192-11e9-9b7c-b842b5e9e592.jpg)

**Comment by jcupitt** [comment]
Try hovering the mouse over one of the red boxes, you should see an error
message in tooltip. What’s the message?

**Comment by bdrichards** [comment]
Here is the message, also, I was mistaken, using the "arrayjoin" command directly from a vips command prompt is successful (I just have hundreds of images to join in dozens of rows...).....
![NIP2-message-2019-03-08_11-37-19](https://user-images.githubusercontent.com/18133188/54051618-146bff80-4197-11e9-8fd7-7a52a28d7c8e.jpg)

**Comment by bdrichards** [comment]
In order to move forward, I uninstalled and installed a previous version (8.6.0) and images load OK. Now, it's time to test with the 30,000 jpg folder of images!!!

**Comment by jcupitt** [comment]
Ouch, 30,000!

Is this a regular grid of images? I'd do it at the command-line, and in a couple of stages. 

Windows can't do more than 2000 files at once, so I'd write a tiny batch script and join up rows 1-10, 11-20, etc. as a set of temporary strip images, then in a second pass join the strips together to make the final output.

**Comment by bdrichards** [comment]
Yes, a regular grid (that's why I think it should be straightforward task to "recombine" them into the original image)!  The folder of image "tiles" is basically the bottom of a tile pyramid ("deep zoom format") and we need to reconstruct the original image (our colleague that created the tile pyramid from the original image has passed away suddenly/unexpectedly and left no hint of where the original large images are - if they even still exist).
Now, you say I should get 2000 files at a time? Ha! I can only manage to get Nip2 to process around 200 successfully at a time - any more than that and the program exits/closes/"stopsworking".
How does one create the command-line to process a folder (or subfolder) of images?
I tried to accomplish that with the VIPS command line and, finding no documentation about how to provide wildcard image paths, tried the following command line versions (this should be straightforward, but how does one specify the folder of images?):
![vips-command](https://user-images.githubusercontent.com/18133188/54074708-9f5a0200-424a-11e9-9dfe-03eb983941f4.JPG)

**Comment by jcupitt** [comment]
Oh dear, I'm sorry to hear about your colleague.

The original win command-line is pretty bad. PowerShell is much better, but I don't know it well, unfortunately.

In linux, you'd do something like:

```
vips arrayjoin "$(echo *.png)" joined.tif --across 22
```

If you zip up the images and put them on dropbox, I could join then for you, if you like.

**Comment by bdrichards** [comment]
Thanks for your generous offer! I have to see if my college's dropbox account has enough space for that! But we may have to take you up on that offer, at least to show "proof-of-concept"
ALSO, good news, the powershell command for a a wildcard list of images functions as below:
![vipsarrayjoin](https://user-images.githubusercontent.com/18133188/54076297-01246700-425f-11e9-8a05-ff2e567786eb.JPG)

Thank you for the suggestion!! (notice the change of "echo" to "dir"). HOWEVER, the dir command returns a list collated incorrectly for what is needed: filename "0-10.jpg" collates BEFORE "0-2.jpg" - yikes, soooooooo, I had to use an advanced renaming utility and add padding to the digits. HOWEVER, again, this increases the size of the name of the file and that limit lowers the number of images that can be processed. Maybe Windows  VIPS can handle 1000's images, but the command line still has limited number of characters, so the length of the individual filenames impacts how many rows of images that can be "tiled together" at a time.....
Thanks again for the suggestion(s) and offer for assistance. Any suggestions on how to overcome the limit on the number of images that get processed? As stated above, I think we are hitting the limit of the command line processor in reality....
